libPhoneNumber-GeocodingParser/libPhoneNumber-GeocodingParser/NBSQLiteDatabaseConnection.m

```diff
@@ -0,0 +1,109 @@
+//
+//  NBSQLiteDatabaseConnection.m
+//  libPhoneNumber-GeocodingParser
+//
+//  Created by Rastaar Haghi on 7/1/20.
+//  Copyright © 2020 Google LLC. All rights reserved.
+//
+
+#import "NBSQLiteDatabaseConnection.h"
+
+@implementation NBSQLiteDatabaseConnection {
+ @private
+  NSURL *_databasePath;
+  sqlite3 *_DB;
+  sqlite3_stmt *_insertStatement;
+  int _sqliteDatabaseCode;
+}
+
+static NSString *const insertPreparedStatement = @"INSERT INTO geocodingPairs%@"
+                                                 @"(NATIONALNUMBER, DESCRIPTION)"
+                                                 @"VALUES "
+                                                 @"(?, ?)";
+
+static NSString *const createTablePreparedStatement =
+    @"CREATE TABLE IF NOT EXISTS geocodingPairs%@ (ID INTEGER PRIMARY KEY "
+    @"AUTOINCREMENT, "
+    @"NATIONALNUMBER TEXT, DESCRIPTION TEXT)";
+static NSString *const createIndexStatement = @"CREATE INDEX IF NOT EXISTS nationalNumberIndex ON "
+                                              @"geocodingPairs%@(NATIONALNUMBER)";
+
+- (instancetype)initWithCountryCode:(NSString *)countryCode
+                       withLanguage:(NSString *)language
+                withDestinationPath:(NSURL *)destinationPath {
+  self = [super init];
+  if (self != nil) {
+    NSString *databasePath = [[NSString alloc]
+        initWithString:[NSString stringWithFormat:@"%@/%@.db", destinationPath, language]];
```

Comment:
Have you tried this. I don't think this is going to work. You need to call `destinationPath.path` to get the actual file path from a NSURL object.

Reply:
worked before, but made this edit for code clarity.